tools/online_editor/src/index.ts

```diff
@@ -380,19 +401,40 @@ export Demo := Window {
       type: "module",
     }
   );
-  lsp_worker.onmessage = m => {
+
+  lsp_worker.onmessage = (m) => {
     // We cannot start sending messages to the client before we start listening which
     // the server only does in a future after the wasm is loaded.
     if (m.data === "OK") {
-
-      const reader = new BrowserMessageReader(lsp_worker);
       const writer = new BrowserMessageWriter(lsp_worker);
 
+      const reader = new ProxyReader(
+        new BrowserMessageReader(lsp_worker),
+        (data: Message) => {
+          if ((data as RequestMessage).method == "slint/load_file") {
```

Comment:
Are we sure the data is always a RequestMessage,  what about the notification and other messages?  don't we need to check for that?

Reply:
There are other message types, but none we care for (so far;-).

This is a type assertion: It tells the compiler "treat `data` as if it was a `RequestMessage`". So in that if-branch I can use all the attributes defined for a `RequestMessage`. That is all this does and in fact this is not even visible in the generated JS code at all.

If this is anything but a `RequestMessage` then requesting `method` from it will most likely return `undefined`, which is not equal to "slint/load_file". So this is save to do as long as no other object happens to also have a `method` set that returns any of the strings we care for... According to the typescript definitions there is no such object.